packages/components/src/alignment-matrix-control/utils.tsx

```diff
@@ -55,12 +69,15 @@ export function getItemId( prefixId, value ) {
 /**
  * Retrieves the alignment index from a value.
  *
- * @param {string} alignment Value to check.
+ * @param  alignment Value to check.
  *
- * @return {number} The index of a matching alignment.
+ * @return The index of a matching alignment.
  */
-export function getAlignmentIndex( alignment = 'center' ) {
+export function getAlignmentIndex(
+	alignment: AlignmentMatrixControlValue = 'center'
+) {
 	const item = transformValue( alignment ).replace( '-', ' ' );
+	if ( ! checkIsValidAlighment( item ) ) return;
```

Comment:
This new type guard (hat tip to @ciampo) will hopefully improve stability/resiliency (so we don't run `indexOf` on an invalid alignment) without introducing any actual change at runtime, as `getAlignmentIndex` would have returned `undefined` anyway.